Sources/StreamChat/Controllers/PollController/PollController.swift

```diff
@@ -0,0 +1,291 @@
+    public func removePollVote(
+        voteId: String,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        pollsRepository.removePollVote(
+            messageId: messageId,
+            pollId: pollId,
+            voteId: voteId,
+            completion: completion
+        )
+    }
+
+    /// Closes the poll.
+    ///
+    /// - Parameters:
+    ///   - completion: A closure to be called upon completion, with an optional `Error` if something went wrong.
+    public func closePoll(completion: ((Error?) -> Void)? = nil) {
+        pollsRepository.closePoll(pollId: pollId, completion: completion)
+    }
+
+    /// Suggests a new option for the poll.
+    ///
+    /// - Parameters:
+    ///   - text: The text of the suggested poll option.
+    ///   - position: An optional position for the suggested option.
+    ///   - custom: An optional dictionary of custom data for the suggested option.
+    ///   - completion: A closure to be called upon completion, with an optional `Error` if something went wrong.
+    public func suggestPollOption(
+        text: String,
+        position: Int? = nil,
+        extraData: [String: RawJSON]? = nil,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        pollsRepository.suggestPollOption(
+            pollId: pollId,
+            text: text,
+            position: position,
+            custom: extraData,
+            completion: completion
+        )
+    }
+    
+    // MARK: - private
+    
+    private func startObserversIfNeeded() {
+        guard state == .initialized else { return }
+        do {
+            try pollObserver?.startObserving()
+            try ownVotesObserver.startObserving()
+            state = .localDataFetched
+        } catch {
+            state = .localDataFetchFailed(ClientError(with: error))
+            log.error("Failed to perform fetch request with error: \(error). This is an internal error.")
+        }
+    }
+}
+
+/// Represents the visibility of votes in a poll.
+public struct VotingVisibility: RawRepresentable, Equatable {
+    public let rawValue: String
+
+    public init(rawValue: String) {
+        self.rawValue = rawValue
+    }
+    
+    /// Votes are public and can be seen by everyone.
+    public static let `public` = Self(rawValue: "public")
+    /// Votes are anonymous and cannot be attributed to individual users.
+    public static let anonymous = Self(rawValue: "anonymous")
+}
+
+extension PollController {
+    struct Environment {
+        var pollObserverBuilder: (
+            _ isBackgroundMappingEnabled: Bool,
+            _ database: DatabaseContainer,
+            _ fetchRequest: NSFetchRequest<PollDTO>,
+            _ itemCreator: @escaping (PollDTO) throws -> Poll,
+            _ fetchedResultsControllerType: NSFetchedResultsController<PollDTO>.Type
+        ) -> EntityDatabaseObserverWrapper<Poll, PollDTO> = {
```

Comment:
Shorter `= EntityDatabaseObserverWrapper.init` should also work here but so important. Same with the other observer.